kubescaler/scaler/aws/cluster.py

```diff
@@ -672,6 +894,28 @@ def scale(self, count):
         # to work. Instead we update the node count and then wait for the nodes.
         # waiter = self.cf.get_waiter('stack_update_complete')
         # waiter.wait(StackName=self.workers_name)
+        self.wait_for_stack_updates()
+        # If successful, save new node count
+        self.node_count = count
+        self.wait_for_nodes()
+        return response
+
+    def scale_using_eks_nodegroup(self, count):
```

Comment:
For scale, do the same pattern you've done with the others - have the logic determined based on the self.eks_nodegroup variable (and that would be the entire content of scale, to return one function or the other, eg.,)

```python
if self.eks_nodegroup:
    return self._scale_using_eks_nodegroup(count)
return self._scale_using_cloudformation()
```
In words, I would put the current content of scale into something like self._scale_using_cloudformation() and then have this be self._scale_using_eks_nodegroup. Let's make them private functions so the user knows not to call directly, but to use scale.

Reply:
Done.